[New Concept Exercise on `lambda`]: Add Secrets draft (#2589)
exercism/python: exercises/concept/secrets/.docs/introduction.md

# Lambdas

> Python has a [`lambda`][lambdas] expression, which is a shorter way to create functions than using `def`. It looks like this:

Reviewer: Maybe something here about them being "nameless" or anonymous?

> ```python

Reviewer: Assigning a `lambda` to a variable is a [Python Anti-pattern](https://docs.quantifiedcode.com/python-anti-patterns/correctness/assigning_a_lambda_to_a_variable.html)

add_one = lambda num: num + 1
```

> In a nutshell, a lambda starts with the keyword `lambda`, then the arguments (separated by a comma, as in a normal `def` function), a colon, and the return value. As they are simply a shorter way of defining a function than `def`, they can be executed in the same way (e.g. `add_one(3)`).

Reviewer: ```suggestion
In short, a `lambda expression` starts with the keyword `lambda`, followed by parameters (_separated by a comma, as you would dow with a `def`-defined function_), a colon, and a `return` value.
```
